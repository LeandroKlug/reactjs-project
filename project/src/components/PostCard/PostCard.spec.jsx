import { render, screen } from '@testing-library/react';
import { PostCard } from ".";
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
        
        render(<PostCard {...props} />);

        expect(screen.getByRole('img', { name: /title/i })).toHaveAttribute('src', props.cover);
        expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument();
        expect(screen.getByText('body1')).toBeInTheDocument();

        // const {debug} = render(<PostCard {...mock} />);
        // debug(); debugando o componente
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard {...props} />);  
        expect(container.firstChild).toMatchSnapshot();
    });
});