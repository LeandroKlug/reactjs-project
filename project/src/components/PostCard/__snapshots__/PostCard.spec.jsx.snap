// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h3>
      title1
    </h3>
    <p>
      body1
    </p>
  </div>
</div>
`;
